### PR TITLE
amppkg_dl_sxg: add -cert_url_path to override cert-url

### PR DESCRIPTION
**Pull request by shigeki**
During renewing of SXG cert in cluster servers, the new cert cannot
be obtained in requesting cert-url from the cluster. `-cert_url_path`
option enables us to overriding scheme, hostname and path of cert-url
so that we can obtain the newer cert by specifying a server.

In the server, we can get a new cert and confirm its cert renewal by
using localhost such as

`amppkg_dl_sxg -cert_url_path http://localhost:8080/amppkg/cert/ http://localhost:8080/priv/doc?...`

### CODE REVIEW

File: cmd/amppkg_dl_sxg/main.go [hunk 1]
```diff
@@ -17,6 +19,7 @@ import (
 
 var flagOutSXG = flag.String("out_sxg", "test.sxg", "Path to where the signed-exchange should be saved.")
 var flagOutCert = flag.String("out_cert", "test.cert", "Path to where the cert-chain+cbor should be saved.")
+var flagCertURL = flag.String("cert_url_path", "", "Override scheme, hostname and path in cert-url.")
```
Comment (twifkak) [inline cmd/amppkg_dl_sxg/main.go]: Rename to cert_url_base. (Maybe s/path/parent path/ in the flag description?)

File: cmd/amppkg_dl_sxg/main.go [hunk 2]
```diff
@@ -90,11 +99,25 @@ func main() {
 	if err != nil {
 		log.Fatalf("%+v", err)
 	}
+	cURL, err := url.Parse(certURL)
+	if err != nil {
+		log.Fatalf("%+v", err)
+	}
+	fURL, err := url.Parse(*flagCertURL)
```
Comment (twifkak) [inline cmd/amppkg_dl_sxg/main.go]: Maybe move this url.Parse call inside the `if *flagCertURL`?

File: cmd/amppkg_dl_sxg/main.go [hunk 3]
```diff
@@ -90,11 +99,25 @@ func main() {
...
+	if err != nil {
+		log.Fatalf("%+v", err)
+	}
+	if *flagCertURL != "" {
+		cURL.Scheme = fURL.Scheme
```
Comment (twifkak) [inline cmd/amppkg_dl_sxg/main.go]: How about something like:

```go
certHash := path.Base(cURL.Path)
cURL = fURL
cURL.Path = path.Join(cURL.Path, certHash)
```

That way, the user can provide a `user:pass@` or `?query` too. (Admittedly, rare cases.)

File: cmd/amppkg_dl_sxg/main.go (hunk 1 above)
Comment (shigeki) [reply cmd/amppkg_dl_sxg/main.go]: Fixed.

File: cmd/amppkg_dl_sxg/main.go (hunk 2 above)
Comment (shigeki) [reply cmd/amppkg_dl_sxg/main.go]: Fixed.

File: cmd/amppkg_dl_sxg/main.go (hunk 3 above)
Comment (shigeki) [reply cmd/amppkg_dl_sxg/main.go]: Fixed.

**Comment by shigeki** [comment]
Thanks for your review. Fixes were done according to your comments.

**Comment by twifkak** [comment]
Thanks, merging.

**Pull request merged**